kolibri/core/utils/portal.py

```diff
@@ -0,0 +1,63 @@
+import json
+import socket
+
+import requests
+from morango.api.serializers import CertificateSerializer
+from morango.certificates import Certificate
+from morango.certificates import Nonce
+from morango.constants.api_urls import NONCE
+from six.moves.urllib.parse import urljoin
+
+from kolibri.core.auth.constants.morango_scope_definitions import FULL_FACILITY
+from kolibri.utils import conf
+
+
+def claim(token, facility, client_ip=None):
+
+    # get client ip if not passed in
+    if not client_ip:
+        hostname = socket.gethostname()
+        client_ip = socket.gethostbyname(hostname)
+
+    # request the server for a one-time-use nonce
+    PORTAL_URL = conf.OPTIONS["Urls"]["DATA_PORTAL_SYNCING_BASE_URL"]
```

Comment:
Perhaps better to have this be the default, but allow it to be overridden as an argument (also in the management command).

Reply:
so in the case there are more data portal servers?

Reply:
Discussed: In the case of other portal servers, use envvar